src/monitord/rotate_log.c

```diff
@@ -285,39 +290,51 @@ void remove_old_logs_m(const char * base_dir, int year, int month, time_t thresh
             tm.tm_mday = day;
 
             if (mktime(&tm) <= threshold) {
-                snprintf(path, PATH_MAX, "%s/%s", base_dir, dirent->d_name);
-                mdebug2("Removing old log '%s'", path);
-                unlink(path);
+                size = snprintf(path, sizeof(path), "%s/%s", base_dir, dirent->d_name);
+
+                if (size + 1 <= (int)sizeof(path)) {
```

Comment:
```suggestion
                if (size < (int)sizeof(path)) {
```
For consistency with other similar lines :)